packages/suite/src/middlewares/wallet/storageMiddleware.ts

```diff
@@ -265,6 +265,7 @@ const storageMiddleware = (api: MiddlewareAPI<Dispatch, AppState>) => {
                 case SUITE.TOGGLE_FIRMWARE_HASH_CHECK:
                 case SUITE.EVM_CONFIRM_EXPLANATION_MODAL:
                 case SUITE.EVM_CLOSE_EXPLANATION_BANNER:
+                case SUITE.SET_IS_COINS_FILTER_VISIBLE:
```

Comment:
this only changes the facdt that settings is saved right after this action is dispatched.

Is the key really persisted? Like, it should be persistent even without this change.

Reply:
yeah, so I tested it out and it **is** persisted alredy, however, it is not saved directly when the filter visibility is toggled.

So you are basically fixing a bug.